### ISSUE TITLE
Multi-Paragraph Quote Highlighting

### ISSUE THREAD

**Issue by ben-albon** [vkbo/novelWriter#546]
I personally find **quote highlighting** to be one of the most appealing features of novelWriter, but it doesn't seem to correctly handle multi-paragraph quotes.

In a multi-paragraph quote, to indicate the original speaker is still talking, the first paragraph is left open: no closing quotation marks are placed.  Without those closing quotation marks, novelWriter does not highlight the first paragraph of the quote.

The expected behaviour is that all quoted text in both the first, and all subsequent paragraphs, would be highlighted.

**Comment by vkbo** [comment]
Yes, this is not really a bug. Markdown doesn't generally allow formatting tags to span multiple lines. The Qt syntax highlighter also applies highlighting to only one line of text at a time (called a text block in Qt), and by default has no knowledge of the line above or below. These two combined is why that particular feature doesn't span paragraphs (or lines if you add line breaks withing a paragraph).

In principle, hanging quotes can be supported by considering a line break as a valid (temporary) closing quote. However, that only works if the writer follows the convention of using a new opening quote mark for the following paragraph(s) and leave them hanging until the end of the quote. Unfortunately, that can only work with double quotes (assuming English language quote style) as the highlighter would be confused by hanging single quoted paragraphs with apostrophes in the text.

I can certainly add support for hanging quotes, but I will then make them optional. You'd probably want to switch them off for single quotes for English, but for other languages where single quote and apostrophe are distinct symbols, you may want it enabled. You can also use the alternate unicode symbol for apostrophe available from the Insert menu. The symbol is not the recommended one by style guides, but in most fonts it renders exactly the same as an English closing single quote.

**Comment by vkbo** [comment]
Ok, so I've tested allowing highlighting for dialogue with no closing quote, following the rules described at the end of [this masterclass article](https://www.masterclass.com/articles/how-to-format-dialogue-in-your-novel-or-short-story). It seems to be working just fine, and I've added options for enabling/disabling this feature to the Preferences. Due to the possible conflict with the apostrophe, I've added separate options for single and double quotes.

I'll make a pull request to the `dev` branch later today, so the feature should be available in the development version then, and part of 1.1 RC1 when that's ready.
